unicode-data-names/lib/Unicode/Char/General/Names.hs

```diff
@@ -0,0 +1,88 @@
+
+{-# LANGUAGE CPP #-}
+
+-- |
+-- Module      : Unicode.Char.General.Names
+-- Copyright   : (c) 2022 Composewell Technologies and Contributors
+-- License     : Apache-2.0
+-- Maintainer  : streamly@composewell.com
+-- Stability   : experimental
+--
+-- Unicode character names and name aliases.
+-- See Unicode standard 14.0.0, section 4.8.
+
+-- [TODO] @since
+
+module Unicode.Char.General.Names
+    ( -- * Name
+      name
+    , nameOrAlias
+      -- * Name Aliases
+    , NameAliases.NameAliasType(..)
+    , nameAliases
+    , nameAliasesByType
+    , nameAliasesByTypes
+    ) where
+
+import Control.Applicative ((<|>))
+import Foreign.C.String (CString)
+import GHC.Exts (Ptr(..), Char(..), chr#, isTrue#, indexInt8OffAddr#, (==#), (+#))
+#if MIN_VERSION_base(4,16,0)
+import GHC.Exts (int8ToInt#)
+#endif
+
+import qualified Unicode.Internal.Char.UnicodeData.DerivedName as DerivedName
+import qualified Unicode.Internal.Char.UnicodeData.NameAliases as NameAliases
+
+-- [TODO] @since
+-- | Name of a character, if defined.
+{-# INLINE name #-}
+name :: Char -> Maybe String
```

Comment:
If we use CString instead of String then the consumer of the APIs could choose to convert that into a format of their choice. Though I am not against using String as well because we can immediately convert the String to another format, that will incur some additional CPU cost but I am not sure if a small difference in performance actually matters for these APIs. In my opinion it probably does not matter much and we can keep the type as String to keep it simple. If it really matters to someone we can later provide lower level APIs that would return CString. 

I just wanted to be sure that you have deliberately chosen `String` here over `CString` and not accidentally or by oversight.

Reply:
We can expose the functions of the internal module here, and rename the current one with suffix "String" (or any other convention).

Reply:
The reasoning to expose Internal modules is:

1. Be able to use any internal APIs in tests
2. Those who wish to experiment without concern about stability of the APIs can use the internal APIs.

Internal APIs can change without notice, therefore, as per PVP if you are using internal APIs then you should depend on the exact minor version of the package.

We can keep it as it is for now. If we feel the need to expose CString APIs as stable APIs later on we can expose those with different names.